common/src/str.rs

```diff
@@ -44,10 +44,23 @@ impl StringReplaceExt for String {
             found = true
         }
 
+        // Make sure that the string is really still valid utf-8 which could
+        // have been violated by the unsafe block. This should always be the
+        // case because str::find is operating on UTF8, so the found positions
+        // should be at the start of UTF8 codepoints and not in the middle.
+        assert!(string_is_really_utf8(self));
+
         found
     }
 }
 
+fn string_is_really_utf8(string: &str) -> bool {
```

Comment:
nit: this naming is a bit weird. Can we just call it `is_utf8` ? The really doesn't seem necessary and it's nice when boolean methods start with is or has or something else clearly indicating a boolean return type in the name.

Reply:
I like the "exceptional" name to guide the reader towards this function being special, because usually an is_utf8 function on strings doesn't make sense as the comment explains.
I am not very attached to this name though so let me know if you still prefer plain `utf-8` and I'll change it.